projects/payment-lib/src/lib/components/reports/reports.component.ts

```diff
@@ -186,7 +191,40 @@ downloadReport(){
           }
         })
 
-    } else {
+    } else if(selectedReportName === 'TELEPHONY_PAYMENTS') {
+
+            this.paymentViewService.downloadTelephonyPaymentsReport(selectedStartDate,selectedEndDate).subscribe(
+              response =>  {
+                this.errorMessage = this.errorHandlerService.getServerErrorMessage(false, false, '');
+
+                const result = {data: JSON.parse(response)['telephony_payments_report_list']};
+                let res= {data: this.applyDateFormat(result)};
+                //res.data= telephonyPaymentsRptDefault;
+                if(result['data'].length > 0) {
+                   for( var i=0; i< res['data'].length; i++) {
+                     if(res['data'][i]["Amount"] !== undefined) {
```

Comment:
Ollie in the future you can map/ foreach any array instead of using indexs such us "i"
1) Map
let transformedData = res['data'].map((item: any) => {
  return /* your transformation logic here */;
});  

2) for each 

res['data'].forEach((item: any) => {
  console.log(item);  // Access each item in the array
});

I will approve it but give me a call I can show you in the future how to avoid indexs